data/data.go

```diff
@@ -18,6 +24,78 @@ type Rule struct {
 	Condition  string        `json:"condition"`
 }
 
+// Serialize builds the Yara rule
+func (rule *Rule) Serialize(output io.Writer) {
+	if rule.Modifiers.Global {
+		fmt.Fprintf(output, "global ")
+	} else if rule.Modifiers.Private {
+		fmt.Fprintf(output, "private ")
+	}
+
+	fmt.Fprintf(output, "rule %s ", rule.Identifier)
+	if len(rule.Tags) > 0 {
+		fmt.Fprintf(output, ": %s ", strings.Join(rule.Tags, " "))
+	}
+
+	fmt.Fprintf(output, "{ \n")
+	if len(rule.Meta) > 0 {
+		fmt.Fprintf(output, "  meta:\n")
+		for _, meta := range rule.Meta {
+			if _, ok := meta.Val.(string); ok {
+				fmt.Fprintf(output, "    %s = \"%s\"\n", meta.Key, meta.Val)
+			}
+			if _, ok := meta.Val.(int64); ok {
+				fmt.Fprintf(output, "    %s = %d\n", meta.Key, meta.Val)
+			}
+			if val, ok := meta.Val.(bool); ok {
+				if val {
+					fmt.Fprintf(output, "    %s = true\n", meta.Key)
+				} else {
+					fmt.Fprintf(output, "    %s = false\n", meta.Key)
+				}
+			}
+		}
+		fmt.Fprintf(output, "\n")
+	}
+
+	if len(rule.Strings) > 0 {
+		fmt.Fprintf(output, "  strings:\n")
+		for _, s := range rule.Strings {
+			if s.Type == TypeString {
+				fmt.Fprintf(output, "    %s = \"%s\"", s.ID, s.Text)
+			} else if s.Type == TypeRegex {
+				fmt.Fprintf(output, "    %s = /%s/", s.ID, s.Text)
+			} else if s.Type == TypeHexString {
+				fmt.Fprintf(output, "    %s = { %s }", s.ID, s.Text)
+			}
+			if s.Modifiers.ASCII {
+				fmt.Fprintf(output, " ascii")
+			}
+			if s.Modifiers.Wide {
+				fmt.Fprintf(output, " wide")
+			}
+			if s.Modifiers.Nocase {
+				fmt.Fprintf(output, " nocase")
+			}
+			if s.Modifiers.Fullword {
+				fmt.Fprintf(output, " fullword")
+			}
+
+			if s.Modifiers.I {
```

Comment:
The `i` and `s` modifiers should go directly after the regex string, for example:

```yara
rule regex_modifiers {
strings:
    $r1 = /regex with modifiers/is
condition:
    any of them
}